db/db_impl_compaction_flush.cc

```diff
@@ -731,6 +756,8 @@ Status DBImpl::CompactRange(const CompactRangeOptions& options,
           continue;
         }
         output_level = level;
+        // only use max_sst_file_number for bottom level compaction
+        max_sst_file_number_to_use = max_sst_file_number;
```

Comment:
I think we should pick a better name than `max_sst_file_number_to_use ` and `max_sst_file_number`. I had a hard time understanding the logic.

Reply:
renamed to `next_file_number` and also made default to `kMaxUint64` so hopefully it's easier to understand